WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -989,60 +990,86 @@ private extension InsightStoreState {
 // MARK: - iOS 14 Widgets Data
 private extension InsightStoreState {
 
-    private func storeHomeWidgetTodayData(stats: TodayWidgetStats) {
-
+    private func storeHomeWidgetData<T: HomeWidgetData>(widgetType: T.Type, stats: Codable) {
         guard #available(iOS 14.0, *),
               let siteID = SiteStatsInformation.sharedInstance.siteID else {
             return
         }
 
-        var homeWidgetTodayCache = HomeWidgetTodayData.read() ?? initializeHomeWidgetTodayData()
-
-        guard let oldData = homeWidgetTodayCache[siteID.intValue] else {
-            DDLogError("HomeWidgetToday: Failed to find a matching site")
+        var homeWidgetCache = T.read() ?? initializeHomeWidgetData(type: widgetType)
+        guard let oldData = homeWidgetCache[siteID.intValue] else {
+            DDLogError("StatsWidgets: Failed to find a matching site")
             return
         }
         let blogService = BlogService(managedObjectContext: ContextManager.shared.mainContext)
 
         guard let blog = blogService.blog(byBlogId: siteID) else {
-            DDLogError("HomeWidgetToday: the site does not exist anymore")
+            DDLogError("StatsWidgets: the site does not exist anymore")
             // if for any reason that site does not exist anymore, remove it from the cache.
-            homeWidgetTodayCache.removeValue(forKey: siteID.intValue)
-            HomeWidgetTodayData.write(items: homeWidgetTodayCache)
+            homeWidgetCache.removeValue(forKey: siteID.intValue)
+            T.write(items: homeWidgetCache)
             return
         }
-        // refresh stats and update any blog info, if they had changed
-        homeWidgetTodayCache[siteID.intValue] = HomeWidgetTodayData(siteID: siteID.intValue,
-                                                                    siteName: blog.title ?? oldData.siteName,
-                                                                    iconURL: blog.icon ?? oldData.iconURL,
-                                                                    url: blog.url ?? oldData.url,
-                                                                    timeZone: blogService.timeZone(for: blog),
-                                                                    date: Date(),
-                                                                    stats: stats)
+        var widgetKind = ""
+        if widgetType == HomeWidgetTodayData.self, let stats = stats as? TodayWidgetStats {
 
-        HomeWidgetTodayData.write(items: homeWidgetTodayCache)
-        WidgetCenter.shared.reloadTimelines(ofKind: WPHomeWidgetTodayKind)
-    }
+            widgetKind = WPHomeWidgetTodayKind
 
-    private func initializeHomeWidgetTodayData() -> [Int: HomeWidgetTodayData] {
+            homeWidgetCache[siteID.intValue] = HomeWidgetTodayData(siteID: siteID.intValue,
```

Comment:
As we discussed on Slack, it would be great if we can avoid needing to know about the concrete data types here, but perhaps something for a future enhancement :)